lib/mllparty/connection_hub/client_monitor.ex

```diff
@@ -0,0 +1,101 @@
+defmodule MLLParty.ConnectionHub.ClientMonitor do
+  use GenServer
+  @name __MODULE__
+
+  @moduledoc """
+    Module to proactively monitor a client's connection and reconnect as necessary
+    Send slack notification if connection goes down for too long, send another when it comes back up
+  """
+
+  require Logger
+
+  alias MLLParty.ConnectionHub.ClientWrapper
+
+  @impl true
+  def init(_opts), do: {:ok, %{
+    down_at: nil,
+    down_notification_sent: false
+  }}
+
+  def start_link() do
+    GenServer.start_link(__MODULE__, [], name: @name)
```

Comment:
@mfos239 GenServer names must be unique on a given node, so passing `__MODULE__` as the GenServer process name here limits the system to only being able to run a single `ClientMonitor` process. 

This manifests as a bug/crash if you attempt to connect to 2 endpoints.

To reproduce:
(in terminal 1):
```
mix phx.server
```

(in terminal 2):
```
iex -S mix
iex> {:ok, r4090} = MLLP.Receiver.start(port: 4090, dispatcher: MLLP.EchoDispatcher)
iex> {:ok, r4091} = MLLP.Receiver.start(port: 4091, dispatcher: MLLP.EchoDispatcher)
```

(in termal 3):
```
curl <send a message to localhost:4090>
curl <send a message to localhost:4091>
```

The 2nd curl will attempt to establish a new connection, which will call `MLLParty.ConnectionHub.ClientMonitor.start_monitor()` a 2nd time, which will cause a pattern match failure on line 27:
```
{:ok, pid} = start_link()
```
because `start_link()` will return `{:error, {:already_started, _some_pid}}`.

A couple of options to mitigate:
- Give monitor processes a unique name
- Move monitor functionality into ClientWrapper process
- Make the the monitor process a single globally unique process (within the node) responsible for monitoring all client connections